open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java

```diff
@@ -0,0 +1,79 @@
+import org.odpi.openmetadata.accessservices.glossaryview.rest.GlossaryViewEntityDetailResponse;
+import org.odpi.openmetadata.commonservices.ffdc.InvalidParameterHandler;
+import org.odpi.openmetadata.commonservices.ffdc.exceptions.InvalidParameterException;
+import org.odpi.openmetadata.commonservices.ffdc.rest.FFDCRESTClient;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException;
+
+/**
+ * Responsible to provide some functionality to requests, like paging and error handling
+ * Actual request is handled higher in the hierarchy
+ */
+class OmasClient extends FFDCRESTClient {
+
+    private InvalidParameterHandler invalidParameterHandler = new InvalidParameterHandler();
+    private GlossaryViewClientExceptionHandler exceptionHandler = new GlossaryViewClientExceptionHandler();
+
+    protected OmasClient(String serverName, String serverPlatformRootURL, String userId, String password) throws
+            org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException{
+        super(serverName, serverPlatformRootURL, userId, password);
+    }
+
+    /*
+     * Meant to send a request to Glossary View Omas that returns a single entity in the response
+     */
+    protected GlossaryViewEntityDetailResponse getSingleEntityResponse(String methodName, String path, String serverName,
+                                                                     String userId, String entityGUID)
+            throws PropertyServerException, InvalidParameterException, GlossaryViewOmasException {
+
+        invalidParameterHandler.validateUserId(userId, methodName);
+        invalidParameterHandler.validateGUID(entityGUID, "entityGUID", methodName);
+
+        GlossaryViewEntityDetailResponse response = callGetRESTCall(methodName, GlossaryViewEntityDetailResponse.class,
+                serverPlatformURLRoot + path, serverName, userId, entityGUID);
+
+        exceptionHandler.detectAndThrowOMRSException(methodName, response);
+
+        return response;
+    }
+
+    /*
+     * Meant to send a request to Glossary View Omas that returns multiple related entities in the response. Supports pagination
+     */
+    protected GlossaryViewEntityDetailResponse getMultipleRelatedEntitiesPagedResponse(String methodName, String path, String serverName,
+                                                                                     String userId, String entityGUID, Integer from, Integer size)
+            throws PropertyServerException, InvalidParameterException, GlossaryViewOmasException{
+
+        invalidParameterHandler.validateUserId(userId, methodName);
+        invalidParameterHandler.validateGUID(entityGUID, "entityGUID", methodName);
+
+        GlossaryViewEntityDetailResponse response = callGetRESTCall(methodName, GlossaryViewEntityDetailResponse.class,
+                serverPlatformURLRoot + path, serverName, userId, entityGUID, from, size);
+
+        exceptionHandler.detectAndThrowOMRSException(methodName, response);
```

Comment:
Not all exceptions are from OMRS - infact there should be NO OMRS exceptions returned to an OMAS client.  The exceptions returned should be common services, OMAS or OCF exceptions.

Reply:
Changed method name to reflect the change of exception type